internal/log/plain.go

```diff
@@ -18,31 +19,57 @@ var colours = map[Level]string{
 
 var _ Sink = (*plainSink)(nil)
 
-func newPlainSink(w io.Writer) *plainSink {
+func newPlainSink(w io.Writer, logTime bool) *plainSink {
 	var isaTTY bool
 	if f, ok := w.(*os.File); ok {
 		isaTTY = isatty.IsTerminal(f.Fd())
 	}
 	return &plainSink{
-		isaTTY: isaTTY,
-		w:      w,
+		isaTTY:    isaTTY,
+		w:         w,
+		startTime: time.Now(),
+		logTime:   logTime,
 	}
 }
 
 type plainSink struct {
-	isaTTY bool
-	w      io.Writer
+	isaTTY    bool
+	w         io.Writer
+	startTime time.Time
+	logTime   bool
 }
 
 // Log implements Sink
 func (t *plainSink) Log(entry Entry) error {
 	var prefix string
+
+	// Add timestamp if required
+	if t.logTime {
+		d := entry.Time.Sub(t.startTime)
+		var timestamp string
+		switch {
+		case d < time.Second:
+			timestamp = fmt.Sprintf("%dms", d/time.Millisecond)
+		case d < time.Minute:
+			timestamp = fmt.Sprintf("%ds", d/time.Second)
+		case d < time.Hour:
+			timestamp = fmt.Sprintf("%dm", d/time.Minute)
```

Comment:
Mmmm so I think this minute/hour granularity will not be super useful in practice. The rationale is that you will still want to be able to tell the _relative_ difference between two log entries...

Reply:
I don't really have a great idea on how to format that though, to be honest.

Reply:
```
     999ms
       20s
   10m 30s
1h 10m 30s
```
This could be more useful?

Reply:
I think it would be okay to drop the spaces.

Though I wonder if at this point it shouldn't just be:

```
  0.999
     20
1:10:30
```

WDYT?

Reply:
`1h10m30s` feels cramped to me in a way that `1:10:30` doesn't

Reply:
I do tend to agree 👍

Reply:
Maybe even `0:20` for seconds. A number by itself is a bit contextless
```
  0.999
   0:20
1:10:30
```

Reply:
At this point maybe we should just use the 24h time rather than relative?

```
16:07:23 info: foo
16:09:01 debug: bar
```

Reply:
Thinking it through... I think relative only makes sense if we mostly cared about startup times, but engineers using ftl would have `ftl dev` running in the background through their day right? And when they look back at the logs the relative timestamps will be in the hours and not be easily convertible to "oh that was 30 mins ago", where as with non relative times it'll be pretty intuitive

Reply:
Updated the code to 24h timestamps